Fail to build iwasm after #1928 merged

I followed the instructions in [WASR Document - Docker Environment Preparation](https://wamr.gitbook.io/document/basics/getting-started/devcontainer). In the container, my toolchain version is as follows:

- cmake version 3.25.2
- gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
- clang version 15.0.7

When I try to build iwasm vmcore, I get this error:

``` plain
refcount.h:132:2: error: "Reference counter isn't implemented"
```

I tried reverting commit 17f3375, which temporarily fixed the issue. I don't understand why this problem occurs, can someone help me?

Full cmake log attached: [cmake.log](https://github.com/bytecodealliance/wasm-micro-runtime/files/10702665/cmake.log)

--- a/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
+++ b/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
@@ -113,7 +113,7 @@ so we have to handle this case specially */
 #if defined(__clang__)
 /* Clang provides stdatomic.h since 3.6.0
 See https://releases.llvm.org/3.6.0/tools/clang/docs/ReleaseNotes.html */
-#if __clang_major__ > 3 || (__clang_minor__ == 3 && __clang_minor__ >= 6)
+#if __clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 6)
 #define CONFIG_HAS_STD_ATOMIC 1
 #else
 #define CONFIG_HAS_STD_ATOMIC 0

--- a/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
+++ b/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
@@ -113,7 +113,7 @@ so we have to handle this case specially */
 #if defined(__clang__)
 /* Clang provides stdatomic.h since 3.6.0
 See https://releases.llvm.org/3.6.0/tools/clang/docs/ReleaseNotes.html */
-#if __clang_major__ >= 3 || (__clang_minor__ == 3 && __clang_minor__ >= 6)
+#if __clang_major__ > 3 || (__clang_minor__ == 3 && __clang_minor__ >= 6)
 #define CONFIG_HAS_STD_ATOMIC 1
 #else
 #define CONFIG_HAS_STD_ATOMIC 0

--- a/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
+++ b/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
@@ -108,11 +108,15 @@
 #endif
 
 #if !defined(BH_PLATFORM_LINUX_SGX)
+/* Clang's __GNUC_PREREQ macro has a different meaning than GCC one,
+so we have to handle this case specially */
 #if defined(__clang__)
 /* Clang provides stdatomic.h since 3.6.0
 See https://releases.llvm.org/3.6.0/tools/clang/docs/ReleaseNotes.html */
 #if __clang_major__ >= 3 || (__clang_minor__ == 3 && __clang_minor__ >= 6)
 #define CONFIG_HAS_STD_ATOMIC 1
+#else
+#define CONFIG_HAS_STD_ATOMIC 0
 #endif
 #elif defined(__GNUC_PREREQ)
 /* Even though older versions of GCC support C11, atomics were

--- a/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
+++ b/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/ssp_config.h
@@ -108,7 +108,13 @@
 #endif
 
 #if !defined(BH_PLATFORM_LINUX_SGX)
-#if defined(__GNUC_PREREQ)
+#if defined(__clang__)
+/* Clang provides stdatomic.h since 3.6.0
+See https://releases.llvm.org/3.6.0/tools/clang/docs/ReleaseNotes.html */
+#if __clang_major__ >= 3 || (__clang_minor__ == 3 && __clang_minor__ >= 6)
+#define CONFIG_HAS_STD_ATOMIC 1
+#endif
+#elif defined(__GNUC_PREREQ)
 /* Even though older versions of GCC support C11, atomics were
 not implemented until 4.9. See
 https://gcc.gnu.org/bugzilla/show_bug.cgi?id=58016 */

